themes/default/content/learn/embedding-pulumi/using-the-api/index.md

```diff
@@ -0,0 +1,348 @@
+    basic_pulumi.refresh_stack(stack=stack)
+    results = basic_pulumi.update_stack(stack=stack)
+    basic_pulumi.destroy_stack(stack=stack, destroy=True)
+    return results
+
+
+# Greeting resource
+class Randomizer(object):
+    def on_get(self, req, resp):
+        payload = {
+            "response": "hello, world"
+        }
+        resp.text = json.dumps(payload)
+        resp.status = falcon.HTTP_200
+
+    def on_get_id(self, req, resp):
+        result = spin_up_program('id')
+        payload = {
+            'response': f'{result.outputs["id"].value}'
+        }
+        resp.text = json.dumps(payload)
+        resp.status = falcon.HTTP_200
+
+    def on_get_uuid(self, req, resp):
+        result = spin_up_program('uuid')
+        payload = {
+            'response': f'{result.outputs["uuid"].value}'
+        }
+        resp.text = json.dumps(payload)
+        resp.status = falcon.HTTP_200
+
+    def on_get_string(self, req, resp):
+        result = spin_up_program('string')
+        payload = {
+            'response': f'{result.outputs["string"].value}'
+        }
+        resp.text = json.dumps(payload)
+        resp.status = falcon.HTTP_200
+
+
+# Instantiation
+randomizer = Randomizer()
+
+# Routes
+api.add_route('/id', randomizer, suffix='id')
+api.add_route('/uuid', randomizer, suffix='uuid')
+api.add_route('/string', randomizer, suffix='string')
+
+if __name__ == '__main__':
+    with make_server('', 8000, api) as httpd:
+        print('Serving on port 8000...')
+
+        # Serve until process is killed
+        httpd.serve_forever()
+
+```
+
+This file sets up a small Falcon application that, when we reach certain
+endpoints, runs our Pulumi program using the commands we made that wrap the CLI.
+Each endpoint calls a different part of the program. Speaking of that program,
+we need to modify it a bit. Change the `__main__.py` file in the
+`burner-program` directory to match this code:
+
+```python
+import pulumi
+import pulumi_random as random
+
+config = pulumi.Config()
+request = config.require('request')
+
+
+def pulumi_program():
+    if request == 'string':
+        fake_string = random.RandomString("fake-string",
+                                          length=24
+                                          )
+        result_string = fake_string.result
+    elif request == 'id':
+        fake_id = random.RandomId("fake-id",
+                                  byte_length=1)
+        result_string = fake_id.b64_std
+    elif request == 'uuid':
+        fake_uuid = random.RandomUuid("fake-uuid")
+        result_string = fake_uuid.result
+    else:
+        result_string = 'Give me something I can work with.'
+    return result_string
+
+
+pulumi.export(f'{request}', pulumi_program())
+
+```
+
+Now, let's run it! From the root of the repo, run `python api.py`. You'll find
+the following output:
+
+```bash
+Serving on port 8000...
+```
+
+Now, we'll try curling the `id` endpoint. Open a new terminal and run this
```

Comment:
Should "curling" be changed to "CURLing"? Probably a stylistic choice, doesn't really matter